plugin/nginx_plus_collectd.py

```diff
@@ -772,6 +801,25 @@ def get_processes(self):
         '''
         return self._send_get(self.processes_url)
 
+    def detect_api_type(self):
+        '''
+        Detects the API type of the Nginx-plus(Newer or Legacy) and initialize the endpoints accordingly
+        '''
+        detected_api_type = self._get_api_type()
+        if self.api_type is not None:
+            if detected_api_type == self.api_type:
+                return
+            else:
+                LOGGER.warning('api type change detected, using \'%s\' API', detected_api_type)
```

Comment:
Why are we ignoring user input?

Reply:
At the initialization time, we are [verifying](https://github.com/signalfx/collectd-nginx-plus/pull/8/files/51b9af9cd2c345fefd0de429ed797587d9e438ad#diff-41f65a45d2a93ce70b773b899a3958f5d64bb2a279351004ffeba0b7d143151cR697) whether the user has provided the correct `api_type` or not, so this function just detects the `api_type` and the verification is being done in the `init` method. Also, At the run time, if the API version gets changed (legacy->newer or version update) our code will detect that and will continue to work with a proper [warning](https://github.com/signalfx/collectd-nginx-plus/pull/8/files/51b9af9cd2c345fefd0de429ed797587d9e438ad#diff-41f65a45d2a93ce70b773b899a3958f5d64bb2a279351004ffeba0b7d143151cR813) message.

Reply:
If a user is manually configuring something invalid we should probably raise it and exit instead of ignoring them and assuming they wanted something else imo.

Reply:
Hello @rmfitzpatrick ,

I completely agree with your above-mentioned point but, as per the latest design(explained [here](https://github.com/signalfx/collectd-nginx-plus/pull/8#discussion_r757691244)), this entire function has been removed. Hence, no changes are required.

Reply:
Speaking to high level functionality this is still a concern.

Reply:
Here, if you mean that we should break the code in case of NGINX+ API version change (r24 -> r25 or any other transition), then we are doing the same in the current logic.